core/templates/services/math-interactions.service.ts

```diff
@@ -171,11 +170,62 @@ export class MathInteractionsService {
       this.warningText = 'Your answer contains an invalid character: "_".';
       return false;
     }
-    if (expressionString.match(/(\+$)|(\+\))/g)) {
+    if (expressionString.match(/(\d\+)|\(\d\+\)/g)) {
+      this.warningText = (
+        'Your answer seems to be missing a number after' +
+        ' the + operator.');
+      return false;
+    }
+
+    if (expressionString.match(/(\+$)|(\(\+\))/g)) {
+      this.warningText = (
+        'Your answer seems to be missing a number before' +
+        ' the + operator.');
+      return false;
+    }
+
+    if (expressionString.match(/\d\*$|\(\d\*\)/g)) {
+      this.warningText = (
+        'Your answer seems to be missing a number after' +
+        ' the × operator.');
+      return false;
+    }
+
+    if (expressionString.match(/(\*$)|(\(\*\))|(\(\*\d\))|(\*\d)/g)) {
       this.warningText = (
-        'Your answer seems to be missing a variable/number after the "+".');
+        'Your answer seems to be missing a number before' +
+        ' the × operator.');
       return false;
     }
+
+    if (expressionString.match(/\d\/$|\(\d\/\)/g)) {
+      this.warningText = (
+        'Your answer seems to be missing a number after' +
+        ' the ÷ operator.');
+      return false;
+    }
+
+    if (expressionString.match(/(\/$)|(\(\/\))|(\(\/\d\))|(\/\d)/g)) {
+      this.warningText = (
+        'Your answer seems to be missing a number before' +
+        ' the ÷ operator.');
+      return false;
+    }
+
+    if (expressionString.match(/\d\-$|\(\d\-\)/g)) {
+      this.warningText = (
+        'Your answer seems to be missing a number after' +
+        ' the - operator.');
+      return false;
+    }
+
+    if (expressionString.match(/(\-$)|(\(\-\))/g)) {
+      this.warningText = (
+        'Your answer seems to be missing a number before' +
+        ' the - operator.');
+      return false;
+    }
+
```

Comment:
@kevintab95 PTAL! 
If these changes are acc to expectations. This code is giving correct error message  only unit test part was failing, please lmk so that I can write unit test further o/w it was blocking my progress.

Reply:
Hi @adityanarayanm095, could you please provide more information about what unit test is failing -- perhaps a debugging doc would help? Also, from what I can tell, a bunch of checks seem to be added in this service but there are no changes to the unit tests in the PR. Should there be accompanying tests in the PR to verify that the functionality works as expected?

Reply:
> Hi @adityanarayanm095, could you please provide more information about what unit test is failing -- perhaps a debugging doc would help? Also, from what I can tell, a bunch of checks seem to be added in this service but there are no changes to the unit tests in the PR. Should there be accompanying tests in the PR to verify that the functionality works as expected?

@kevintab95 I am just asking here that is this code okay So that I can proceed further with writing unit test part?